lib/src/repo.rs

```diff
@@ -1252,7 +1302,59 @@ mod tests {
     use super::*;
 
     #[test]
-    fn test_id_index() {
+    fn test_id_index_resolve_prefix() {
+        fn sorted(resolution: PrefixResolution<Vec<i32>>) -> PrefixResolution<Vec<i32>> {
+            match resolution {
+                PrefixResolution::SingleMatch(mut xs) => {
+                    xs.sort(); // order of values might not be preserved by IdIndex
+                    PrefixResolution::SingleMatch(xs)
+                }
+                _ => resolution,
+            }
+        }
+        let id_index = IdIndex::from_vec(vec![
+            (ChangeId::from_hex("0000"), 0),
+            (ChangeId::from_hex("0099"), 1),
+            (ChangeId::from_hex("0099"), 2),
+            (ChangeId::from_hex("0aaa"), 3),
+            (ChangeId::from_hex("0aab"), 4),
+        ]);
+        assert_eq!(
+            id_index.resolve_prefix_with(&HexPrefix::new("0").unwrap(), |&v| v),
+            PrefixResolution::AmbiguousMatch,
+        );
+        assert_eq!(
+            id_index.resolve_prefix_with(&HexPrefix::new("00").unwrap(), |&v| v),
+            PrefixResolution::AmbiguousMatch,
+        );
+        assert_eq!(
+            id_index.resolve_prefix_with(&HexPrefix::new("000").unwrap(), |&v| v),
+            PrefixResolution::SingleMatch(vec![0]),
+        );
+        assert_eq!(
+            id_index.resolve_prefix_with(&HexPrefix::new("0001").unwrap(), |&v| v),
+            PrefixResolution::NoMatch,
+        );
+        assert_eq!(
+            sorted(id_index.resolve_prefix_with(&HexPrefix::new("009").unwrap(), |&v| v)),
+            PrefixResolution::SingleMatch(vec![1, 2]),
```

Comment:
Added explicit `sort()` here because I use `.sort_unstable_by()` to sort entries by key, though there wouldn't be no runtime randomness.